### DISCUSSION TITLE
Using spacy's NER to for Entity Recognition of custom entities

### DISCUSSION THREAD

**Discussion by shrinidhin** [explosion/spaCy#9940] [Help: Coding & Implementations] [answered]
Hi!So I am currently working on building a custom entity recognizer using spacy. Whilst preparing the training examples, For tagging the data using the BILOU scheme, should I only add the new custom entity tags in my data while keeping the tags for the other tokens as 'O', or should I tag those as well? I only want to add new entities to the existing NER.

**Comment by polm** [comment, marked as answer]
It sounds like the best thing is to label just your entities and train a separate NER model to combine with the existing one. See the [double NER](https://github.com/explosion/projects/tree/v3/tutorials/ner_double) example for notes on how that works.

Also note that while you say you want to add the new entity to the existing NER model, I would recommend against that. If you only add your new entities as training data you'll run into the "catastrophic forgetting" problem, so you'd have to add data for all entities you want to keep using. But that would be a lot of work, can be hard to get right, and having all the entities in one model isn't necessarily better than having separate models.

**Reply by shrinidhin** [reply]
So if I take the first route, I will essentially have to label all entities old as well as new in my training data?If I want to train my NER model on my labelled training data and combine with spacy's existing NER.

**Reply by polm** [reply]
I am not sure what you mean by "the first route". My suggestion is you label *just* your new entities and use that to train a model. Then you can combine that model with the existing one as a separate component. The example project does something similar.

**Reply by shrinidhin** [reply]
Got it. I'm sure this will solve my problem.Thanks a lot!

**Reply by arshad171** [reply]
Apart from the size or the size/RAM considerations mentioned in the link above, are there any other aspects that need to be considered with the above approach?
Will the approach of merging NER models (by adding components) work well for any number of mutually exclusive set of NER models?

**Reply by polm** [reply]
I think the RAM considerations are pretty important. Besides that, if you have a lot of models, I'd be suspicious of whether they were actually independent.

**Reply by arshad171** [reply]
yeah, the labels are mutually exclusive. and if there are related labels, I add it to the existing NER model and rehearse the model (pseudo-rehearsal).
